qemu/tests/timerdevice_time_jump_check.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+from virttest import error_context
+from virttest.compat_52lts import decode_to_text
+
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    check time jumps in guest (only for Linux guest):
+
+    1) boot guest with '-rtc base=utc,clock=host,driftfix=slew'
+    2) check current clocksource in guest
+    3) pin all vcpus to specfic host CPUs
+    4) verify time jump
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    error_context.context("Check the clock source currently used on guest",
+                          logging.info)
+    cmd = "cat /sys/devices/system/clocksource/"
+    cmd += "clocksource0/current_clocksource"
+    logging.info("%s is current clocksource." % session.cmd_output(cmd))
+
+    error_context.context("Pin every vcpu to physical cpu", logging.info)
+    host_cpu_cnt_cmd = params["host_cpu_cnt_cmd"]
+    host_cpu_num = decode_to_text(process.system_output(host_cpu_cnt_cmd,
+                                                        shell=True)).strip()
+    host_cpu_list = (_ for _ in range(int(host_cpu_num)))
+    cpu_pin_list = list(zip(vm.vcpu_threads, host_cpu_list))
+    if len(cpu_pin_list) < len(vm.vcpu_threads):
+        test.error("There isn't enough physical cpu to"
+                   " pin all the vcpus")
+
```

Comment:
@sitoliu  If vcpu number >physical number,  os supports multiple vcpu pin to one physical cpu.   Must it one to one?

Reply:
@xiangchunfu Thanks, update it.